### PR TITLE
Backend for KCF custom images

### PR DESCRIPTION
**Pull request by sdreyer**
# Description

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: provider/k8s.go [hunk 1]
```diff
@@ -390,11 +385,25 @@ func NewLocalExecutor(config Config, logger *zap.SugaredLogger) (Executor, error
...
 }
 
-func (kube KubernetesExecutor) isDefaultImage() bool {
+type KubernetesExecutor struct {
+	logger *zap.SugaredLogger
+	image  string
+}
+
+func (kube *KubernetesExecutor) isDefaultImage() bool {
 	return strings.Contains(kube.image, cfg.PandasBaseImage)
```
Comment (simba-git) [inline provider/k8s.go:394]: This is technically not correct right? You could have a base image that contains the same name as the default image. I'd recommend just leaving a comment about that and leaving it as is.

File: provider/k8s.go [hunk 2]
```diff
@@ -1061,18 +1070,14 @@ func (k8s *K8sOfflineStore) pandasRunnerArgs(outputURI string, updatedQuery stri
...
-	// }
 	sourceList := strings.Join(sources, ",")
 	envVars := map[string]string{
 		"OUTPUT_URI":          outputURI,
 		"SOURCES":             sourceList,
 		"TRANSFORMATION_TYPE": "df",
 		"TRANSFORMATION":      code,
 	}
-	_, ok := k8s.executor.(KubernetesExecutor)
+	_, ok := k8s.executor.(*KubernetesExecutor)
```
Comment (simba-git) [inline provider/k8s.go:1080]: You could merge this with the line below

**Review by simba-git — APPROVED** [review]
Minor changes

File: provider/k8s.go (hunk 1 above)
Comment (sdreyer) [reply provider/k8s.go:394]: Added

File: provider/k8s.go (hunk 2 above)
Comment (sdreyer) [reply provider/k8s.go:1080]: Changed the other one as well

**Pull request merged**